BenchmarkSample/ui/src/main/java/com/example/benchmark/ui/SampleAdapter.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.benchmark.ui
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+
+class SampleAdapter : ListAdapter<String, Holder>(StringDiffCallback) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
+        return Holder.create(parent)
+    }
+
+    override fun onBindViewHolder(holder: Holder, position: Int) {
+        holder.bind(getItem(position))
+    }
+}
+
+private object StringDiffCallback : DiffUtil.ItemCallback<String>() {
+    override fun areItemsTheSame(oldItem: String, newItem: String): Boolean =
```

Comment:
add comment along the lines of:

in a real diff callback, you'd want to return true if two objects refer to same object, and should e.g. crossfade. In this sample, we don't have changing list item, so simply use equality